rust/driver_manager/src/error.rs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt;
+
+use adbc_core::error::Status;
+
+/// A Driver Manager error.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Error {
+    Adbc(adbc_core::error::Error),
+    DynamicLibrary(String),
+    Other(String),
+}
```

Comment:
I think we should figure out how to unify the `adbc_core` errors with the driver manager errors. It's weird to have an "adbc error" that needs to be extended to contain errors from the "adbc driver manager". It's all ADBC.

Reply:
Thank you! Much clearer this way.